src/Datadog.Trace.ClrProfiler.Managed/Integrations/Types/TypeExtensions.cs

```diff
@@ -0,0 +1,29 @@
+namespace Datadog.Trace.ClrProfiler
+{
+    internal static class TypeExtensions
+    {
+        public static System.Type GetInstrumentedType(
+            this object runtimeObject,
+            string instrumentedTypeName)
+        {
+            if (runtimeObject == null)
+            {
+                return null;
+            }
+
+            var currentType = runtimeObject.GetType();
+
+            while (currentType != null)
```

Comment:
This will never be null because `GetType()` never returns null.

Reply:
It's re-assigned to the `BaseType` below.
https://docs.microsoft.com/en-us/dotnet/api/system.type.basetype

Reply:
I could convert to a do while.

Reply:
> t's re-assigned to the BaseType below.
Oh, I missed that. Never mind!